radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokenclasses/BurnTokensActionMapper.java

```diff
@@ -63,22 +70,23 @@ public BurnTokensActionMapper(RadixUniverse universe, Function<RadixAddress, Obs
 
 		List<SpunParticle> particles = new ArrayList<>();
 
-		long consumerTotal = 0;
-		final long subUnitAmount = burnTokensAction.getAmount().multiply(TokenClassReference.getSubUnits()).longValueExact();
+		BigInteger consumerTotal = BigInteger.ZERO;
+		final BigInteger subUnitAmount = burnTokensAction.getAmount().multiply(TokenClassReference.getSubUnits()).toBigIntegerExact();
 		Iterator<OwnedTokensParticle> iterator = unconsumedOwnedTokensParticles.iterator();
-		Map<ECKeyPair, Long> newUpQuantities = new HashMap<>();
+		Map<ECKeyPair, UInt256> newUpQuantities = new HashMap<>();
 
 		// HACK for now
 		// TODO: remove this, create a ConsumersCreator
 		// TODO: randomize this to decrease probability of collision
-		while (consumerTotal < subUnitAmount && iterator.hasNext()) {
-			final long left = subUnitAmount - consumerTotal;
+		while (consumerTotal.compareTo(subUnitAmount) < 0 && iterator.hasNext()) {
+			final BigInteger left = subUnitAmount.subtract(consumerTotal);
```

Comment:
Why are we using `BigInteger` instead of `UInt256` here?

Reply:
I was concerned about overflow here, but on a closer look, it seems to be OK.  I've changed to use UInt256.